nohup: ignoring input
 * Serving Flask app 'app'
 * Debug mode: on
[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://192.168.100.226:5000
[33mPress CTRL+C to quit[0m
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 223-125-772
127.0.0.1 - - [24/Sep/2025 15:50:17] "[32mGET / HTTP/1.1[0m" 302 -
127.0.0.1 - - [24/Sep/2025 15:50:26] "GET /auth/login HTTP/1.1" 200 -
127.0.0.1 - - [24/Sep/2025 15:50:26] "[32mPOST /auth/login HTTP/1.1[0m" 302 -
127.0.0.1 - - [24/Sep/2025 15:50:26] "GET /dashboard HTTP/1.1" 200 -
127.0.0.1 - - [24/Sep/2025 15:50:26] "GET /dashboard HTTP/1.1" 200 -
 * Detected change in '/home/victor/Documents/Code/pos-system/config/config.py', reloading
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 223-125-772
127.0.0.1 - - [24/Sep/2025 15:54:30] "[35m[1mGET /products/purchases HTTP/1.1[0m" 500 -
Traceback (most recent call last):
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/app.py", line 1478, in __call__
    return self.wsgi_app(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/app.py", line 1458, in wsgi_app
    response = self.handle_exception(e)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/app/views/auth.py", line 44, in decorated_function
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/app/views/products.py", line 337, in purchases
    return render_template('products/purchases.html', title='Purchases', purchases=purchases)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/templating.py", line 151, in render_template
    template = app.jinja_env.get_or_select_template(template_name_or_list)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/jinja2/environment.py", line 1087, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/jinja2/environment.py", line 1016, in get_template
    return self._load_template(name, globals)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/jinja2/environment.py", line 975, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/jinja2/loaders.py", line 126, in load
    source, filename, uptodate = self.get_source(environment, name)
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/templating.py", line 65, in get_source
    return self._get_source_fast(environment, template)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/templating.py", line 99, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: products/purchases.html
127.0.0.1 - - [24/Sep/2025 15:54:30] "[36mGET /products/purchases?__debugger__=yes&cmd=resource&f=debugger.js HTTP/1.1[0m" 304 -
127.0.0.1 - - [24/Sep/2025 15:54:30] "[36mGET /products/purchases?__debugger__=yes&cmd=resource&f=style.css HTTP/1.1[0m" 304 -
127.0.0.1 - - [24/Sep/2025 15:54:30] "[36mGET /products/purchases?__debugger__=yes&cmd=resource&f=console.png HTTP/1.1[0m" 304 -
127.0.0.1 - - [24/Sep/2025 15:54:30] "[36mGET /products/purchases?__debugger__=yes&cmd=resource&f=console.png HTTP/1.1[0m" 304 -
127.0.0.1 - - [24/Sep/2025 15:54:37] "[35m[1mGET /products/ HTTP/1.1[0m" 500 -
Traceback (most recent call last):
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/app.py", line 1478, in __call__
    return self.wsgi_app(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/app.py", line 1458, in wsgi_app
    response = self.handle_exception(e)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/app/views/products.py", line 157, in products
    return render_template('products/products.html',
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/templating.py", line 152, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/flask/templating.py", line 133, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/jinja2/environment.py", line 1295, in render
    self.environment.handle_exception()
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/jinja2/environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/victor/Documents/Code/pos-system/app/templates/products/products.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/home/victor/Documents/Code/pos-system/app/templates/base.html", line 288, in top-level template code
    {% block scripts %}{% endblock %}
  File "/home/victor/Documents/Code/pos-system/app/templates/products/products.html", line 383, in block 'scripts'
    'X-CSRFToken': '{{ csrf_token() }}'
  File "/home/victor/Documents/Code/pos-system/venv/lib/python3.12/site-packages/jinja2/utils.py", line 92, in from_obj
    if hasattr(obj, "jinja_pass_arg"):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'csrf_token' is undefined
127.0.0.1 - - [24/Sep/2025 15:54:37] "[36mGET /products/?__debugger__=yes&cmd=resource&f=style.css HTTP/1.1[0m" 304 -
127.0.0.1 - - [24/Sep/2025 15:54:37] "[36mGET /products/?__debugger__=yes&cmd=resource&f=debugger.js HTTP/1.1[0m" 304 -
127.0.0.1 - - [24/Sep/2025 15:54:37] "[36mGET /products/?__debugger__=yes&cmd=resource&f=console.png HTTP/1.1[0m" 304 -
127.0.0.1 - - [24/Sep/2025 15:54:46] "[32mGET / HTTP/1.1[0m" 302 -
127.0.0.1 - - [24/Sep/2025 15:54:46] "GET /dashboard HTTP/1.1" 200 -
127.0.0.1 - - [24/Sep/2025 15:54:46] "GET /api/dashboard/chart-data HTTP/1.1" 200 -
 * Detected change in '/home/victor/Documents/Code/pos-system/app/__init__.py', reloading
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 223-125-772
 * Detected change in '/home/victor/Documents/Code/pos-system/app/__init__.py', reloading
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 223-125-772
 * Detected change in '/home/victor/Documents/Code/pos-system/app/models/__init__.py', reloading
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 223-125-772
 * Detected change in '/home/victor/Documents/Code/pos-system/app/models/__init__.py', reloading
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 223-125-772
